api/helpers/lock-game.js

```diff
@@ -34,15 +34,22 @@ module.exports = {
         const now = dayjs.utc().toDate();
         const lockIsStaleTimeout = dayjs.utc().subtract(30, 'second').toDate();
 
-        // Lock & re-fetch game if unlocked or lock is expired
-        const updatedGame = await Game.updateOne({
+        // Lock game if unlocked or lock is expired
+        let updatedGame = await Game.updateOne({
```

Comment:
Do you find the same game here? I don’t see why you need a let, it should be updated game and refreshed updated game or skipping the variable assignment for the first one altogether since you just need to await the call.

Reply:
The reason I had it this way is because `Model.updateOne()` returns the updated record, but not its associations (in this case `p0` and `p1`). We need the record itself to verify that the lock is set, and what I had was checking that first, and if the lock matches, _then_ querying again to populate the other associations.

What I could do instead would be skip capturing it in the `updateOne()` call itself and then immediately re-query without checking the lock to grab the associations. I think it might read more clearly that way. In the case where we fail to acquire the lock (basically because we hit multiple competing concurrent requests), this means we would be running an extra query for the Game + associations even though we were going to throw it away to try again. I think that's probably okay for clarity's sake